src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/DefaultDocumentWriter.cs

```diff
@@ -176,7 +176,14 @@ public override void VisitMethodDeclaration(MethodDeclarationIRNode node)
 
             public override void VisitFieldDeclaration(FieldDeclarationIRNode node)
             {
-                Context.Writer.WriteField(node.AccessModifier, node.Modifiers, node.Type, node.Name);
+                if (node.Annotations.ContainsKey(CommonAnnotations.InitializeTagHelperVariables))
+                {
+                    Context.TagHelperWriter.WriteDeclareTagHelperFields(Context);
+                }
+                else
+                {
+                    Context.Writer.WriteField(node.AccessModifier, node.Modifiers, node.Type, node.Name);
+                }
```

Comment:
This isn't right at all.

This doesn't semantically change anything, it's just using the file declaration as a proxy.